lib/lotus/mailer.rb

```diff
@@ -94,5 +96,72 @@ def self.load!
     def self.reset!
       configuration.reset!
     end
+
+    module ClassMethods
+      # Delivers a multipart email. It instantiates a mailer and deliver the email.
+      #
+      # @since 0.1.0
+      #
+      # @example email delivery through smtp via gmail configured with environment variables
+      # class DeliveryMethodMailer
+      #   include Lotus::Mailer
+      #
+      #   from ENV["GMAIL_USER"]
+      #   to "inesopcoelho@gmail.com"
+      #   subject "This is the subject"
+      #
+      # end
+      #
+      # MyCustomDeliveryMethod = {
+      #   :address              => "smtp.gmail.com",
+      #   :port                 => 587,
+      #   :domain               => "localhost:8000",
+      #   :user_name            => ENV["GMAIL_USER"],
+      #   :password             => ENV["GMAIL_PASSWORD"],
+      #   :authentication       => "plain",
+      #   :enable_starttls_auto => true
+      # }
+      # Lotus::Mailer.configure do
+      #   delivery_method :smtp, MyCustomDeliveryMethod
+      # end
+      #
+      # DeliveryMethodMailer.deliver
+      def deliver(locals = {})
+        new(locals).deliver
+      end
+    end
+
+    module InstanceMethods
+      # Delivers a multipart email, by looking at all the associated templates and render them.
+      #
+      # @since 0.1.0
+      def deliver
+        mail['from'] = self.class.from
+        mail['to'] = self.class.to
+        mail['subject'] = self.class.subject
+        if !(Lotus::Mailer.configuration.delivery_method.nil?)
```

Comment:
what about `if Lotus::Mailer.configuration.delivery_method`? You could check for truthiness here